### ISSUE TITLE
no-named-default: Should ignore type import

### ISSUE THREAD

**Issue by ilyub** [import-js/eslint-plugin-import#2130]
Sample code:
```ts
import type { default as Vue, VueConstructor } from "vue";
```

Getting error:
Use default import syntax to import 'Vue'

However, this code produces typescript error:
```ts
import type Vue, { VueConstructor } from "vue";
```

And this code triggers import/no-duplicates rule:
```ts
import type { VueConstructor } from "vue";
import type Vue from "vue";
```

So, either:
A) the original code should be ignored because there is no alternative for it.
or
B) import/no-duplicates should allow the last code.

See also:
https://github.com/benmosher/eslint-plugin-import/issues/2114

**Comment by ljharb** [comment]
#2149 fixed no-duplicates; the proper thing is to use two statements (until TS fixes things so it properly accepts them in the same statement).
